flux-refineck/src/constraint_gen.rs

```diff
@@ -63,12 +63,17 @@ struct InferCtxt<'a, 'tcx> {
 #[derive(PartialEq, Eq, Clone, Copy, Hash)]
 pub struct Tag {
     pub reason: ConstrReason,
-    pub span: Span,
+    pub src_span: Span,
+    pub dst_span: Option<ESpan>,
 }
 
 impl Tag {
     pub fn new(reason: ConstrReason, span: Span) -> Self {
-        Self { reason, span }
+        Self { reason, src_span: span, dst_span: None }
+    }
+
+    pub fn dst(self, dst_span: Option<ESpan>) -> Self {
```

Comment:
I'd prefix this with `with_`, the name `dst` suggest that you are extracting the destination from the `Tag`.

```suggestion
    pub fn with_dst(self, dst_span: Option<ESpan>) -> Self {
```